### PR TITLE
Introduce parents

### PR DESCRIPTION
**Pull request by jstastny-cz**
Refactor of the repo structure.

Main themes:
* extracting sources download to separate modules that can be ad-hoc brought into Maven reactor to provide sources needed for testing
* introducing grouping parent for repositories
* the parent modules are the invocation points for tests for given repository (e.g. kogito-runtimes)
  * defining profiles to group child module with the source-downloading module (mentioned above)

### CODE REVIEW

**Review by dupliaka — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
Jan, this is amazing piece of work! Thank you.

File: pom.xml [hunk 1]
```diff
@@ -49,6 +46,12 @@
     <main.basedir>${maven.multiModuleProjectDirectory}</main.basedir>
     <settings.xml.file>${main.basedir}/settings/settings.xml</settings.xml.file>
     <sources.directory>${project.build.directory}/sources/</sources.directory>
+    <sources.directory.module.sources.download>
+      ${main.basedir}/run-tests-with-build-sources-download/target/sources
```
[the remark this answers is not in the document]
Comment (jomarko) [reply pom.xml:50]: I understand already, it is a dependency in some leaves, just not too common for my experience from projects I worked on.

**Review by hellowdan — APPROVED** [review]
Thanks for this!!!

File: run-tests-with-build-optaplanner-parent/optaplanner-project-sources-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-invoker-plugin</artifactId>
+        <configuration>
+          <!-- kogito-runtimes tests were not designed to run in parallel -->
+          <parallelThreads>1</parallelThreads>
```
[the remark this answers is not in the document]
Comment (jstastny-cz) [reply run-tests-with-build-optaplanner-parent/optaplanner-project-sources-test/pom.xml:45]: Yeah, result of copy&paste probably.

**Review by jomarko — APPROVED** [review]
thank you

File: run-tests-with-build-optawebs-parent/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Usage
+Project pom defines profiles that can invoke tests for a particular delivery option.
+Download of artifacts is provided by means described in [Module structure](#module-structure).
+
+### Project-sources testing
+* optaweb-employee-rostering
+  ```
+  mvn clean verify \
+  -Doptawebs.test=project-sources \
```
Comment (radtriste) [inline run-tests-with-build-optawebs-parent/README.md:14]:
```suggestion
  -Dtest.type=project-sources \
```

File: run-tests-with-build-optawebs-parent/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  -Dproject.sources.artifact=optaweb-employee-rostering \
+  -Doptaplanner.download.build.version=<VERSION>
+  ```
+* optaweb-vehicle-routing
+  ```
+  mvn clean verify \
+  -Doptawebs.test=project-sources \
```
Comment (radtriste) [inline run-tests-with-build-optawebs-parent/README.md:21]:
```suggestion
  -Dtest.type=project-sources \
```

**Review by radtriste — CHANGES_REQUESTED** [review]
instead of `optaplanner.download.build.version` and `kogito.download.build.version`, would it make sense to have only one `download.build.version` ?
or we want to keep it safe ?

**Review by radtriste — APPROVED** [review]

**Pull request merged**